code/modules/client/preference_setup/general/01_basic.dm

```diff
@@ -167,4 +206,33 @@
 			pref.metadata = sanitize(new_metadata)
 			return TOPIC_REFRESH
 
+	else if(href_list["ipc_tag"])
+		var/choice = alert(user, "Do you wish for your IPC to have a tag?\n\nWARNING: Being an untagged IPC in Tau space is highly illegal!", "IPC Tag", "Yes", "No")
+		if(CanUseTopic(user))
+			if(choice == "Yes")
+				pref.machine_tag_status = TRUE
+			else
+				pref.machine_tag_status = FALSE
+			return TOPIC_REFRESH
+
+	else if(href_list["serial_number"])
+		var/new_serial_number = sanitize(input(user, "Enter what you want to set your serial number to.", "IPC Serial Number", pref.machine_serial_number) as message|null)
+		new_serial_number = uppertext(dd_limittext(new_serial_number, 12))
+		if(new_serial_number && CanUseTopic(user))
+			pref.machine_serial_number = sanitize(new_serial_number)
+			return TOPIC_REFRESH
+
+	else if(href_list["generate_serial"])
+		if(pref.real_name)
+			var/generated_serial = uppertext(dd_limittext(md5(pref.real_name), 12))
+			pref.machine_serial_number = generated_serial
+			return TOPIC_REFRESH
+
+	else if(href_list["ownership_status"])
+		var/static/list/ownership_options = list(IPC_OWNERSHIP_COMPANY, IPC_OWNERSHIP_PRIVATE, IPC_OWNERSHIP_SELF)
+		var/new_ownership_status = input(user, "Choose your IPC's ownership status.", "IPC Ownership Status") as null|anything in ownership_options
+		if(new_ownership_status && CanUseTopic(user))
+			pref.machine_ownership_status = new_ownership_status
+			return TOPIC_REFRESH
```

Comment:
This should only be possible to be changed at the char creation and a short grace-period afterwards (similar to the name)
After that IC actions (or admin involvement) should be required to change that.

Reply:
Lore team wants it to be changed at will to allow for character freedom. The name restriction is fairly draconic.

Reply:
Discussed with the deputy loremaster, the outlined changes are alright with them.
When implementing this, I would suggest to implement it the same way as the name restrictions, but with a configuration variable that can control the duration.
That way we can have it disabled for a few weeks to allow IPC players to edit their existing characters before we enable the time restrictions.